otellog/otel_log.go

```diff
@@ -0,0 +1,70 @@
+package otellog
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"time"
+)
+
+type Logger interface {
+	Log(level LogLevel, msg fmt.Stringer)
```

Comment:
```suggestion
	Log(ctx context.Context, level LogLevel, msg fmt.Stringer, attrs ...attribute.KeyValue)
	ContextDecoder(ctx context.Context) []attribute.KeyValue
```
What about this interface? I think just including the `Context` is not enough. It should provide an way to "decode" the `Context`, which means getting values from the `Context` and logging these values that we exactly need.
For example, In a http server, the `Context` invocation chain always begins from the `Request.Context()`. Therefore, there are many redundant values in the passed `Context`. It is expensive to log all these values. What we need in such a `Context` may just the tracing related values. In such a situation, including a `ContextDecoder` can solve it easily, remaining scalability and flexibility.
In default case, it is enough maintaining a `NoopDecoder` that ignoring the `Context` and returning nil.